connman/src/service.c

```diff
@@ -8389,10 +8390,12 @@ int __connman_service_connect(struct connman_service *service,
 
 	service->connect_reason = reason;
 
-	if (err >= 0)
+	switch (err) {
+	case 0:
```

Comment:
Previously this function returned 0 here if `err > 0`, but now it will continue instead. I couldn't figure out whether you can assume that `err <= 0`, or if this change is intentional, so I'm flagging it just in case.

Reply:
In any case success should be a return of 0. So this previously did not react to any other negative error than `-EINPROGRESS` and the resulting calls to [service.c:service_connect()](https://github.com/sailfishos/connman/blob/master/connman/src/service.c#L8229)  leading to two: 

1. [network.c:__connman_network_connect()](https://github.com/sailfishos/connman/blob/master/connman/src/network.c#L1533) -> calls to plugin that do not seem to use positive ERRNO values for any other than D-Bus messages, check with:
  ```
  for i in $(errno --list|cut -d' ' -f1) ; do
      find connman/plugins/* -type f -exec grep -e $i {} + |grep -v -e "-$i" -e "DEFAULT"
  done
  ```
or

2. the call to [provider.c:__connman_provider_connect()](https://github.com/sailfishos/connman/blob/master/connman/src/provider.c#L313) -> [plugins/vpn.c:connect_provider()/provider_connect()](https://github.com/sailfishos/connman/blob/master/connman/plugins/vpn.c#L603)

Do return only negative values with error cases and 0 is a-ok. In any ways treating any other than 0 as a mark of success is wrong in this context. I cannot remember a case where that was used as success but there have been issues of handling the non-negative ERRNOs.

This could for sure handle more errors but maybe for later..

Reply:
Thanks for checking all this so thoroughly. I'm reassured that it's okay.